documentation : correspondance externalcode uri

azime opened the pull request.

azime committed:
documentation : correspondance externalcode uri

azime committed:
ed : externalcode vers uri

correspondance entre le code externe navitia 1 et uri navitia 2

azime committed:
connector_at : récupération de l'uri à partir du code externe

azime committed:
ed : correspondance code externe uri

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+
+namespace ed{ namespace connectors{
+
+ExtCode2Uri::ExtCode2Uri(const std::string& redis_connection){
```
Il faudrait plutôt créer une classe d'abstraction générique (au hasard Redis) et la mettre dans utils, de cette façon nous pourrons la réutiliser pour faire autre chose

Pourquoi utiliser une chaîne de caractère puis la spliter et ne pas utiliser plusieurs arguments?

azime replied:
5 arguments de plus, ça va faire trop d'arguments pour gtfs2ed en plus c'est spliter une seule fois.
je peux rajouter les 5 s'il faut !

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+    logger = log4cplus::Logger::getInstance("log");
+}
+
+bool ExtCode2Uri::connected(){
```
Connected devrait retourner si la connexion est ouverte, et non pas l'ouvrir (et accessoirement _is_connected_ serait à préférer)

La gestion des erreurs devrait etre gérer par des exception plutot que des codes de retour: http://google-styleguide.googlecode.com/svn/trunk/cppguide.xml#Exceptions
Comme il le font remarquer, ils n'utilisent pas les exceptions uniquement à cause de l'historique, probléme que nous n'avons pas.

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+            LOG4CPLUS_FATAL(logger, "Connecting redis failed, error : "+std::string(connect->errstr));
+            return false;
+        }
+    }catch(...){
```
les catch ... sont à banir, on attrape tout et n'importe quoi, de plus hiredis étant une lib C, j'aurais tendance à penser qu'elle ne throw pas d'exception

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+        this->db = str_to_int(key_value.at("db"));
+    }
+
+    this->separator = ":";
```
étant donné que nous utilisons deja le ":" dans nos URI il faudrait choisir un autre séparateur, on n'a qu'a prendre des pipes, histoire de garder nos ptites habitudes :D

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+bool ExtCode2Uri::set(const std::string& key, const std::string& value){
+
+    std::string command_set = "SET " + this->prefix + this->separator + key + " " + value;
+     bool to_return;
```
l'indentation n'est pas bonne

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+     bool to_return;
+    try{
+        to_return = redisCommand(this->connect, command_set.c_str());
+    }catch(...){
```
idem, c'est du C, donc pas d'exception et quand bien meme le "..." c'est le mal absolu.

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+
+bool ExtCode2Uri::set(const std::string& key, const std::string& value){
+
+    std::string command_set = "SET " + this->prefix + this->separator + key + " " + value;
```
privilégier quelque chose de cette forme:

```
 reply = redisCommand(context, "SET key:%s %s", myid, value);
```

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+    std::string command_set = "SET " + this->prefix + this->separator + key + " " + value;
+     bool to_return;
+    try{
+        to_return = redisCommand(this->connect, command_set.c_str());
```
redisCommand retourne un pointeur sur la réponse, qu'il faut donc regarder et nettoyer via "freeReplyObject"

kinnou02 commented on source/ed/connectors/extcode2uri.cpp:
```diff
@@ -0,0 +1,120 @@
+    return to_return;
+}
+
+bool ExtCode2Uri::set_lists(const ed::Data & data){
```
Cette fonction ne devrait pas etre dans la classe d'abstraction à hiredis

kinnou02 commented on source/ed/connectors/extcode2uri.h:
```diff
@@ -0,0 +1,34 @@
+namespace ed{ namespace connectors{
+class ExtCode2Uri
+    {
+private:
```
dans l'doute: protected

kinnou02 commented on source/ed/data.h:
```diff
@@ -11,7 +11,7 @@
 template<typename T>
 void normalize_uri(std::vector<T*>& vec){
-    std::string prefix = navitia::type::static_data::get()->captionByType(T::type);
+    std::string prefix = navitia::type::static_data::get()->captionByType(T::type);    
```
attention: trailling space

kinnou02 commented on source/ed/tests/CMakeLists.txt:
```diff
@@ -21,7 +15,11 @@ target_link_libraries(adapted_test ed types connectors pq pqxx data georef
 
+add_executable(extcode2uri_test extcode2uri_test.cpp)
+target_link_libraries(extcode2uri_test ed connectors types utils hiredis ${BOOST_LIBS} log4cplus)
+
```
Il manque le add_test

kinnou02 commented on source/ed/tests/extcode2uri_test.cpp:
```diff
@@ -0,0 +1,13 @@
+
+
+BOOST_AUTO_TEST_CASE(required_files) {
+    ed::connectors::ExtCode2Uri connection("host=localhost db=0 password=navitia port=6379");
```
les information de connexion devrait être récupéré depuis des variables d'environnement, avec une valeur par défaut si pas trouvé

kinnou02 commented on source/ed/connectors/extcode2uri.h:
```diff
@@ -0,0 +1,34 @@
+        ExtCode2Uri(const std::string& redis_connection);
+        ~ExtCode2Uri();
+        bool connected();
+        bool set(const std::string& key, const std::string& value);
```
il faudrait rajouter un get

kinnou02 commented on source/connectors/at/selector/atreader.py:
```diff
@@ -69,20 +69,18 @@ def get_navitia_type(object_type):
 
 
-class AtRealtimeReader(object):
+class AtRealtimeReader:
```
Pourquoi supprimer l'héritage de object?

kinnou02 commented on source/connectors/at/daemon.py:
```diff
@@ -38,6 +41,12 @@ def _init_logger(self, filename='', level='debug'):
                 .setLevel(logging.INFO)
 
+    def _init_redisqueue(self):
+        self.redis_queue = ext_code_2_uri.RedisQueue(self.config.redisqueque_host,
```
le module _"ext_code_2_uri_" n'existe pas, et donc lève le point suivant: pourquoi les tests passe si il manque des modules?

kinnou02 commented:
Et la documentation n'est pas à jours

azime committed:
connector_at : ajout du module redis

- Réorganisation du répértoire connectors
- Ajout des tests unitaires pour redis
- Ne pas prendre en comptes les messages n'ayant pas
de correspondances externalcode uri

azime committed:
documentation : correspondance ExternalCode Uri

azime committed:
ed : ajout des correspondances externalcode, URi dans REDIS

azime committed:
ed : correction orthographe

azime committed:
Utils : Ajout des deux fichiers redis

azime committed:
connnector_at : Renvoie de None si l'uri n'existe pas

azime committed:
ed :deplacement du linckage REDIS

azime committed:
Deletion of depency to redis

azime committed:
redis : ne pas envoyer le mot de pas si vide

kinnou02 commented on source/utils:
```diff
@@ -1 +1 @@
-Subproject commit 28bf9d3be6bed2f480852ad2532ce10fd329a26f
+Subproject commit 7a4a79af4f198c2cccd84ce1b26caa4b25e301d8
```
la référence du commit n'est pas bonne, il faut que tu mette à jour utils, ou que tu push tes changements

kinnou02 commented on source/ed/tests/extcode2uri_test.cpp:
```diff
@@ -0,0 +1,66 @@
+
+BOOST_AUTO_TEST_CASE(exception) {
+
+    std::string host("localhost");
```
```
host = "localhost"
```

ça marche tout aussi bien

kinnou02 commented on source/ed/tests/extcode2uri_test.cpp:
```diff
@@ -0,0 +1,66 @@
+BOOST_AUTO_TEST_CASE(test1) {
+
+    std::string host("localhost");
+    std::string db("0");
```
tu devrait faire une fixture por ne pas avoir à copier collé ce bout la à chaque fois

kinnou02 commented on source/ed/tests/extcode2uri_test.cpp:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    ed::connectors::ExtCode2Uri redis_test("host=" + host + " db=" + db + " password=" + password + " port=" + port + " timeout=" + timeout);
+    redis_test.redis.redis_connect_with_timeout();
```
idéalement la connexion devrait être ouvert lors de l'appel au constructeur

kinnou02 commented on source/ed/tests/extcode2uri_test.cpp:
```diff
@@ -0,0 +1,66 @@
+
+    std::string host("localhost");
+    std::string db("0");
+    std::string password("password");
```
par défaut le password doit etre null, à partir de la 2.4 redis refuse la connexion si on lui passe un password alors qu'il n'est pas requis

kinnou02 commented on source/ed/gtfs2ed.cpp:
```diff
@@ -96,6 +98,18 @@ int main(int argc, char * argv[])
+            ext_code_2_uri.to_redis(data);
+        }catch(const navitia::exception& ne){
+            std::cout << "Impossible d'alimenter redis" << std::endl;
+            std::cout << ne.what() << std::endl;
```
dans ce cas la on devrait sortir avec un code différent de 0, le référentiel n'a pas pu etre mis à jours donc on annule le traitement

(idéalement il faudrait rollback la transaction postgres aussi)

azime replied:
A présent rien n'est écrit dans la base postgres

kinnou02 commented on source/ed/connectors/extcode2uri.h:
```diff
@@ -0,0 +1,33 @@
+        try{
+            extcode2uri.redis.set(element->external_code, element->uri);
+        }catch(const navitia::exception& ne){
+            LOG4CPLUS_WARN(extcode2uri.logger, "add_redis : " + std::string(ne.what()));
```
il faudrait re throw l'exception

kinnou02 commented on source/connectors/connectors/redis_queue.py:
```diff
@@ -0,0 +1,29 @@
+
+    def __init__(self, host,port,db,password, prefix='REDIS'):
+        try:
+            self._redis = Redis(host=host,port=port,db=db,password=password)
```
les espaces!!!! :@

kinnou02 commented on source/connectors/connectors/redis_queue.py:
```diff
@@ -0,0 +1,29 @@
+            self._separator = "|"
+        except:
+            print("Redis : Connecting redis failed" )
+            self._redis = None
```
except sans rien = le mal absolu

et de toute facon tu doit raise quelque chose, ton objet est mort si tu n'a pas pu instancier redis

kinnou02 commented on source/connectors/connectors/redis_queue.py:
```diff
@@ -0,0 +1,29 @@
+
+class RedisQueue():
+
+    def __init__(self, host,port,db,password, prefix='REDIS'):
```
met des valeurs par défaut, c'est toujours agréable

kinnou02 commented on source/connectors/connectors/redis_queue.py:
```diff
@@ -0,0 +1,29 @@
+#encoding: utf-8
+from redis import Redis
+
+class RedisQueue():
```
pourquoi RedisQueue?

kinnou02 commented:
faudra que tu rebase depuis dev aussi

azime committed:
connector_at : paramètres de connection par défaut

azime committed:
ed : usage de fixture pour les tests unitaires

azime committed:
ed : génération d'une exception si erreur

azime committed:
ed : interception de l'exception

lors de la construction de l'objet extcode2uri

azime committed:
ed : connection à redis

lors de la construction de l'objet extcode2uri

azime committed:
Utils: update submodule

azime committed:
connector_at : renemmage de RedisQueque en RedisHelper

azime commented:
des corrections sont effectuées suite aux remarques

azime committed:
connector_at : revue des normes de codage

kinnou02 commented:
Il faut que tu rebase depuis dev pour que le merge puisse etre effectué

azime committed:
connector_at : Ajout des messages dans le logger

au lieu de print et suppression des espaces

azime committed:
connector_at : Message in english

azime committed:
ed : suppression de l'exception

azime commented:
rebase externalcode2uri
